menas/src/main/scala/za/co/absa/enceladus/menas/repositories/VersionedMongoRepository.scala

```diff
@@ -132,9 +132,17 @@ abstract class VersionedMongoRepository[C <: VersionedModel](mongoDb: MongoDatab
       set("userDisabled", username))).toFuture()
   }
 
-  def setLockState(name: String, isLocked: Boolean): Future[UpdateResult] = {
+  def setLockState(name: String, isLocked: Boolean, username: String): Future[UpdateResult] = {
+    val additionalFields = if (isLocked) {
+      (ZonedDateTime.now(), username)
+    } else {
+      (null, null)
+    }
+
     collection.updateMany(getNameFilter(name), combine(
-      set("locked", isLocked))).toFuture()
+      set("locked", isLocked),
+      set("dateLocked", additionalFields._1),
+      set("userLocked", additionalFields._2))).toFuture()
   }
```

Comment:
```suggestion
 def setLockState(name: String, isLocked: Boolean, username: String, datetime: ZonedDateTime = ZonedDateTime.now()): Future[UpdateResult] = {
    val (dateLocked, userLocked) = if (isLocked) {
      (datetime, username)
    } else {
      (null, null)
    }

    collection.updateMany(getNameFilter(name), combine(
      set("locked", isLocked),
      set("dateLocked", dateLocked),
      set("userLocked", userLocked))).toFuture()
  }
```

How about making the now-datetime overridable from param (easier testing) and directly naming the tuple items?